src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -79,7 +80,14 @@ public CommandResult execute(Model model) throws CommandException {
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+        // A person's uniqueness is based on their phone and email. If the edit command results in sharing those
+        // identifiers with 2 others, it means the uniqueness invariant was violated and hence there are duplicates.
+        if (countDuplicatePerson(fullPersonList, editedPerson) == 2) {
```

Comment:
I know this would not happen, but following defensive programming you should account for cases like 3.

Perhaps check for >1

Reply:
Implemented this and below.